### PR TITLE
Harden dependency resolution

### LINKED ISSUE #196: Setup dependabot on psycop-t2d

### PR DESCRIPTION
**Pull request by MartinBernstorff**
Checks are NA, only CI.

## Notes for reviewers
This should help us keep dependencies under control.

Dependabot auto-merge seems pretty neat! Hope it works, will have to test on `main`.

Fixes #196.

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -1,5 +1,6 @@
 - [ ] I have battle-tested on Overtaci (RMAPPS1279)
 - [ ] I have bumped the version as described in the [wiki](https://www.notion.so/Setting-up-Poetry-on-your-local-machine-76647d75e8fa4d9ba553cbc2a49946d9#dca1e0f2ce934a49acd1851aaf882d75)
+- [ ] I have assigned ranges (e.g. `>=0.1, <0.2`) to all new, non-dev dependencies
```
Comment (KennethEnevoldsen) [inline .github/pull_request_template.md:3]: dependabot also updated dev dependencies

File: .github/dependabot.yml [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  - package-ecosystem: "pip" # See documentation for possible values
+    directory: "/" # Location of package manifests
+    schedule:
+      interval: "monthly"
+      day: "monday"
+      time: "05:00"
+      timezone: "Europe/Copenhagen"
+    reviewers: 
+      - "MartinBernstorff"
+    open-pull-requests-limit: 20
```
Comment (KennethEnevoldsen) [inline .github/dependabot.yml:17]:
```suggestion
    open-pull-requests-limit: 10
```

File: .github/workflows/dependabot_automerge.yml [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  dependabot:
+    - name: Enable automerge for Dependabot PRs
+      runs-on: ubuntu-latest
+      if: ${{ github.actor == 'dependabot[bot]' && github.event_name == 'pull_request'}} # Detect that the PR author is dependabot
+      steps:
+        - name: Enable auto-merge for Dependabot PRs
+          run: gh pr merge --auto --merge "$PR_URL" # Use Github CLI to merge automatically the PR
+          env:
+            PR_URL: ${{github.event.pull_request.html_url}}
+            GITHUB_TOKEN: ${{secrets.GITHUB_TOKEN}}
```
Comment (KennethEnevoldsen) [inline .github/workflows/dependabot_automerge.yml:18]: Have you added this (as far as I remember it does not allow "GITHUB_*" tokens)

File: pyproject.toml [hunk 4]
```diff
@@ -14,7 +14,7 @@ plotly = ">=5.8.2,<5.9.0"
 matplotlib = ">=3.5.2,<3.6.0"
 wandb = ">=0.12.21,<0.13.0"
 psycopmlutils = {git = "https://github.com/Aarhus-Psychiatry-Research/psycop-ml-utils.git", rev = "main"}
-tabulate = "^0.8.10"
+tabulate = ">=0.8.10, <0.9.0"
```
Comment (KennethEnevoldsen) [inline pyproject.toml:17]:
```suggestion
tabulate = ">=0.8.10,<0.9.0"
```

File: .github/workflows/dependabot_automerge.yml (hunk 3 above)
Comment (MartinBernstorff) [reply .github/workflows/dependabot_automerge.yml:18]: `By default, GitHub Actions workflows triggered by Dependabot get a GITHUB_TOKEN with read-only permissions.`
https://docs.github.com/en/code-security/dependabot/working-with-dependabot/automating-dependabot-with-github-actions#changing-github_token-permissions

Comment (MartinBernstorff) [reply .github/workflows/dependabot_automerge.yml]: Oh, but this isn't actually a dependabot workflow, so it should get the same permissions as any other.

Looks good to you?

**Review by KennethEnevoldsen — APPROVED** [review]

**Pull request merged**